doc/migrations/v1.0.0-v2.0.0.md

```diff
@@ -0,0 +1,245 @@
+})
+
+// is the connection limited?
+console.info(conn.limits?.seconds) // number | undefined
+console.info(conn.limits?.bytes) // bigint | undefined
+
+// register a protocol handler for incoming protocol dials
+node.handle('/my-protocol/1.0.0', ({ stream, connection }) => {
+  // handle protocol
+}, {
+   runOnLimitedConnection: true
+})
+
+// register a topology that is notified of peers connecting over limited
+// connections
+node.register('/my-protocol/1.0.0', {
+  onConnect: (peer, connection) => {
+    // handle connect
+  },
+  notifyOnLimitedConnection: true
+})
+```
+
+## The CustomEvent polyfill has been removed from `@libp2p/interface`
+
+The `@libp2p/interface` previously exported a [CustomEvent](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent)
+polyfill for compatibility with Node.js.
+
+As of `v18.7.0`, Node.js ships with a global implementation of [CustomEvent](https://nodejs.org/docs/latest/api/globals.html#customevent)
+so the polyfill is no longer necessary and it has been removed.
+
+Please use the global implementation instead.
+
+**Before**
+
+```ts
+import { CustomEvent } from '@libp2p/interface'
+
+const event = new CustomEvent()
+```
+
+**After**
+
+```ts
+const event = new CustomEvent()
+```
+
+## `localPeer` is no longer passed to Connection Encrypters
+
+Implementations of the [ConnectionEncrypter](https://libp2p.github.io/js-libp2p/interfaces/_libp2p_interface.ConnectionEncrypter.html)
+interface should obtain the PeerID of the current node from the components map
+instead of as an argument to the `secureInbound`/`secureOutbound` methods.
+
+See [accessing libp2p components](https://github.com/libp2p/js-libp2p/blob/main/doc/SERVICES.md#accessing-libp2p-components)
+in the guide to writing services.
+
+If you use a connection encrypter like `noise` or `tls`, all you need to do is
+upgrade to the latest version.
+
+## The `.code` property has been removed from thrown errors, use `.name` instead
+
+Errors thrown by libp2p previously had a `.code` property that lets the handler
+differentiate between different error conditions.
+
+The use of this was largely influenced by Node.js' use of `.code` in errors
+thrown by the [fs](https://nodejs.org/docs/latest/api/fs.html) module.
+
+Since then JavaScript has largely coalesced around using the [.name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name)
+property instead.
+
+**Before**
+
+```ts
+try {
+  // a libp2p operation that may throw
+} catch (err: any) {
+  if (err.code === 'ERR_NOT_FOUND') {
+    // handle the error appropriately
+  }
+}
+```
+
+**After**
+
+```ts
+try {
+  // a libp2p operation that may throw
+} catch (err: any) {
+  if (err.name === 'NotFoundError') {
+    // handle the error appropriately
+  }
+}
+```
+
+## pubsub, identify, circuit-relay v2 and record envelope sealing use `privateKey` component
+
+The Peer ID of the current node contains a public/private keypair that is used
```

Comment:
2.0 could be an opportunity to remove the private key from the peer id entirely

Reply:
@wemeetagain is there a tracking issue/discussion for removing the private key from the peer ID?

Reply:
https://github.com/libp2p/js-libp2p/issues/2659

Reply:
#2660